extensions/ql-vscode/package.json

```diff
@@ -663,13 +663,17 @@
         }
       ],
       "commandPalette": [
+        {
+          "command": "codeQL.authenticateToGitHub",
+          "when": "config.codeQL.canary"
+        },
         {
           "command": "codeQL.runQuery",
           "when": "resourceLangId == ql && resourceExtname == .ql"
         },
         {
           "command": "codeQL.runRemoteQuery",
-          "when": "editorLangId == ql && resourceExtname == .ql"
+          "when": "config.codeQL.canary && editorLangId == ql && resourceExtname == .ql"
```

Comment:
Have you verified this is working correctly?

Reply:
Yep, the commands are hidden in the right situations! Annoyingly, the authentication option itself is still visible from the side bar, even with the canary setting `false`...

![image](https://user-images.githubusercontent.com/42641846/122467696-bd2e7980-cfb2-11eb-9533-d31462fa8d46.png)

That comes from the initialization step, which isn't feature-flagged.
https://github.com/github/vscode-codeql/blob/fbec5c920784cdc497b9236135dabfb6b207fb7e/extensions/ql-vscode/src/extension.ts#L729

Will have a look at tidying that up tomorrow! 🧹

Reply:
> Yep, the commands are hidden in the right situations! 

Nice!

> Annoyingly, the authentication option itself is still visible from the side bar, even with the canary setting `false`...

Hmmm...is it sufficient to wrap the call to `Credentials.initialize` in a check to see if canary is set?

Also, if someone changes from canary to not canary, it would be nice if they were logged out of github. You can set up a configuration listener and automatically log out if canary goes from true to false (similarly, you can automatically log in when the setting goes in the other direction).

Reply:
> Also, if someone changes from canary to not canary, it would be nice if they were logged out of github. You can set up a configuration listener and automatically log out if canary goes from true to false (similarly, you can automatically log in when the setting goes in the other direction).

I did some more experimenting, and I don't know if it's possible to log a user out via the public authentication API if we're using a "built-in" [AuthenticationProvider](https://code.visualstudio.com/api/references/vscode-api#AuthenticationProvider) (i.e. GitHub or Microsoft)... I found a few comments where other people were also talking about the [lack of a "logout" option](https://github.com/microsoft/vscode/issues/88309#issuecomment-741569123) and how the [authentication providers are registered](https://github.com/microsoft/vscode/issues/100993#issuecomment-665160371), so I think this might be a wider problem 🤔 

For now, I've at least wrapped the call to `Credentials.initialize` in a canary check, so everything is fully hidden when it needs to be 🎉 Is that okay for now? I don't expect many users toggle the canary flag by accident, so it shouldn't have a noticeable impact!

Reply:
Oh well. Thanks for looking into this and apologies for suggesting something that isn't possible.

It's not a big deal since if someone is already logged in and in canary mode, then they are already aware of this feature and will have tried to use it. And it's sufficient to just hide the feature if canary is off. I'll take another look shortly.